tests/namedtempfile.rs

```diff
@@ -326,3 +338,86 @@ fn test_keep() {
     }
     std::fs::remove_file(&path).unwrap();
 }
+
+#[test]
+fn test_make() {
+    let tmpfile = Builder::new()
+        .make(|path| File::create(path))
+        .unwrap();
+
+    assert!(tmpfile.path().is_file());
+}
+
+#[test]
+fn test_make_in() {
+    let tmp_dir = tempdir().unwrap();
+
+    let tmpfile = Builder::new()
+        .make_in(tmp_dir.path(), |path| File::create(path))
+        .unwrap();
+
+    assert!(tmpfile.path().is_file());
+    assert_eq!(tmpfile.path().parent(), Some(tmp_dir.path()));
+}
+
+#[cfg(unix)]
+#[test]
+fn test_make_uds() {
+    use std::os::unix::net::UnixListener;
+
+    let temp_sock = Builder::new()
+        .prefix("tmp")
+        .suffix(".sock")
+        .rand_bytes(12)
+        .make(|path| UnixListener::bind(path))
```

Comment:
Note: I couldn't get `.make(UnixListener::bind)` to work because of the following error:
```
error: implementation of `FnOnce` is not general enough
   --> tests/namedtempfile.rs:372:10
    |
372 |         .make(UnixListener::bind)
    |          ^^^^ implementation of `FnOnce` is not general enough
    |
    = note: `fn(&'2 Path) -> Result<UnixListener, std::io::Error> {UnixListener::bind::<&'2 Path>}` must implement `FnOnce<(&'1 Path,)>`, for any lifetime `'1`...
    = note: ...but it actually implements `FnOnce<(&'2 Path,)>`, for some specific lifetime `'2`
```

Reply:
Hm. Oh well.